opencensus/exporters/trace/zipkin/internal/zipkin_exporter.cc

```diff
@@ -0,0 +1,365 @@
+    // Failed to set http user agent.
+    std::cerr << curl_easy_strerror(res);
+    return;
+  }
+  if (CURLE_OK != (res = curl_easy_setopt(curl, CURLOPT_POSTFIELDSIZE, size))) {
+    // Failed to set http body size.
+    std::cerr << curl_easy_strerror(res);
+    return;
+  }
+  if (CURLE_OK != (res = curl_easy_setopt(curl, CURLOPT_POSTFIELDS, data))) {
+    // Failed to set http body data.
+    std::cerr << curl_easy_strerror(res);
+    return;
+  }
+  if (CURLE_OK !=
+      curl_easy_setopt(curl, CURLOPT_CONNECTTIMEOUT,
+                       absl::ToInt64Milliseconds(options.connect_timeout))) {
+    // Failed to set connect timeout.
+    std::cerr << curl_easy_strerror(res);
+    return;
+  }
+  if (CURLE_OK !=
+      curl_easy_setopt(curl, CURLOPT_TIMEOUT_MS,
+                       absl::ToInt64Milliseconds(options.request_timeout))) {
+    // Failed to set request timeout.
+    std::cerr << curl_easy_strerror(res);
+    return;
+  }
+  if (CURLE_OK != (res = curl_easy_setopt(curl, CURLOPT_NOSIGNAL, 1))) {
+    // Failed to disable signals.
+    std::cerr << curl_easy_strerror(res);
+    return;
+  }
+
+  if (options.proxy.empty()) {
+    if (CURLE_OK !=
+        (res = curl_easy_setopt(curl, CURLOPT_PROXY, options.proxy.c_str()))) {
+      // Failed to set proxy.
+      std::cerr << curl_easy_strerror(res);
+      return;
+    }
+
+    if (options.http_proxy_tunnel) {
+      if (CURLE_OK !=
+          (res = curl_easy_setopt(curl, CURLOPT_PROXYTYPE, CURLPROXY_HTTP))) {
+        // Failed to set HTTP proxy type.
+        std::cerr << curl_easy_strerror(res);
+        return;
+      }
+      if (CURLE_OK !=
+          (res = curl_easy_setopt(curl, CURLOPT_HTTPPROXYTUNNEL, 1))) {
+        // Failed to set HTTP proxy tunnel.
+        std::cerr << curl_easy_strerror(res);
+        return;
+      }
+    }
+  }
+
+  if (options.max_redirect_times > 0) {
+    if (CURLE_OK != (res = curl_easy_setopt(curl, CURLOPT_FOLLOWLOCATION, 1))) {
+      // Failed to enable follow location.
+      std::cerr << curl_easy_strerror(res);
+      return;
+    }
+    if (CURLE_OK != (res = curl_easy_setopt(curl, CURLOPT_MAXREDIRS,
+                                            options.max_redirect_times))) {
+      // Failed to set max redirect times.
+      std::cerr << curl_easy_strerror(res);
+      return;
+    }
+  }
+
+  // Sending HTTP request to url.
+  if (CURLE_OK != (res = curl_easy_perform(curl))) {
+    // Failed to send http request.
+    std::cerr << curl_easy_strerror(res);
+    return;
+  }
+}
+
+}  // namespace
+
+void ZipkinExporter::SendMessage(const std::string &msg, size_t size) const {
+  struct curl_slist *headers = nullptr;
+  char err_msg[CURL_ERROR_SIZE] = {0};
+  CURL *curl = curl_easy_init();
+
+  if (!curl) {
+    // Failed to create curl handle.
+    return;
+  }
+
+  headers = curl_slist_append(headers, "Content-Type: application/json");
+  headers = curl_slist_append(headers, "Expect:");
```

Comment:
Is this necessary? If so please add a comment to explain.

Reply:
I don't think these are strictly necessary.  Removed.